src/App.tsx

```diff
@@ -17,18 +31,19 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter filter={filter} setFilter={setFilter} setSearchText={setSearchText} searchText={searchText} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {(todos?.length > 0)
```

Comment:
You can actually filter list of todos in this component and pass to `TodoList` already filtered todos. This way you will not need to pass so many props down.